src/Microsoft.DotNet.Build.Tasks/PackageRestoreLogger.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class PackageRestoreLogger : ILogger
+    {
+        public LoggerVerbosity Verbosity { get; set; }
+        public string Parameters { get; set; }
+
+        private HashSet<string> _packageSet = new HashSet<string>();
+        private string _logFile;
+        private bool _outputToConsole = true;
+        private Func<string, bool> _packageUrlCheck = x => x.Contains("OK") || x.Contains("CACHE");
+
+        public void Initialize(IEventSource eventSource)
+        {
+            ParseParameters();
+            eventSource.MessageRaised += eventSource_MessageHandler;
+        }
+
+        private void ParseParameters()
+        {
+            if (!String.IsNullOrEmpty(Parameters))
+            {
+                string[] parameterPairs = Parameters.Split(';');
+                foreach (string parameterPair in parameterPairs)
+                {
+                    if (parameterPair.Length > 0)
+                    {
+                        string[] parameterNameValue = parameterPair.Split('=');
+
+                        ApplyParameter(parameterNameValue[0], parameterNameValue.Length > 1 ? parameterNameValue[1] : null);
+                    }
+                }
+            }
+        }
+
+        private void ApplyParameter(string name, string value)
+        {
+            switch (name.ToLower())
+            {
+                case "logfile":
+                    _logFile = value;
+                    break;
+                case "logtoconsole":
+                    _outputToConsole = Boolean.Parse(value);
+                    break;
+                default:
+                    // ignore unrecognized parameters
+                    break;
+            }
+        }
+
+        private void eventSource_MessageHandler(object sender, BuildMessageEventArgs e)
+        {
+            if (_packageUrlCheck(e.Message))
+            {
+                int index = e.Message.IndexOf("https");
```

Comment:
I thought we were going to get rid of this substring logic

Reply:
Given that you're initially searching for "OK" and "CACHE", I would just trim those instead of doing this search